src/components/structured-list/structured-list.js

```diff
@@ -0,0 +1,94 @@
+import mixin from '../../globals/js/misc/mixin';
+import createComponent from '../../globals/js/mixins/create-component';
+import initComponentBySearch from '../../globals/js/mixins/init-component-by-search';
+import eventMatches from '../../globals/js/misc/event-matches';
+
+class StructuredList extends mixin(createComponent, initComponentBySearch) {
+  /**
+   * StructuredList
+   * @extends CreateComponent
+   * @extends InitComponentBySearch
+   * @param {HTMLElement} element The root element of tables
+   * @param {Object} [options] the... options
+   * @param {string} [options.selectorInit] selector initialization
+   * @param {string} [options.selectorRow] css selector for selected row
+   */
+  constructor(element, options) {
+    super(element, options);
+    this.rows = [...this.element.ownerDocument.querySelectorAll(this.options.selectorRow)];
```

Comment:
Also, I'd suggest *not* caching rows in constructor, because such cache will be stale as soon as dynamic DOM addition happens. (Dynamic DOM addition is one of the most popular Carbon components questions recently)